src/Domain/Creators/BinaryCompactObject.cpp

```diff
@@ -146,6 +153,21 @@ BinaryCompactObject::BinaryCompactObject(
         "of Layer 1 enveloping Object B requires excising the interior of "
         "Object B");
   }
+  if ((inner_boundary_condition_ != nullptr and
+       outer_boundary_condition_ == nullptr) or
+      (inner_boundary_condition_ == nullptr and
+       outer_boundary_condition_ != nullptr)) {
+    PARSE_ERROR(context,
+                "Must specify either both inner and outer boundary conditions "
+                "or neither.");
```

Comment:
If I don't excise A or B (say for a BNS run) then what inner boundary condition do I want to pass? None? Or should we not require the inner BC in this scenario?

Reply:
I don't know how to not require a BC in this case because we would effectively need a "default" value. The only ways to really handle this that I can think of right now are:
1. have an `Alternatives<tmpl::list<ExciseA, ExciseB, InnerBoundaryCondition>, tmpl::list<>>`, but I don't think this will work because this is effectively setting a "default" value
2. leave as-is with a comment
3. decide to add a None boundary condition, either system-specific or general. Probably a general one like the periodic would work

Reply:
My suggestion is to:  
- Add a `None` BC that errors if it is ever used
- Either require that `None` is used if ExciseInteriorA and ExciseInteriorB are both false; or add a comment to InnerBoundaryCondition's help string saying that this option is ignored in that case

Reply:
I would be satisfied with a comment (as per @kidder's suggestion) for now, and perhaps adding a `None` BC in a separate PR (or making an issue for it).